signalfx-transform-helper.js

```diff
@@ -1,36 +1,30 @@
 'use strict';
 
-const FIELD_SEPARATOR = '_';
 const MAX_FIELDNAME_LENGTH = 128;
+const DETAIL_PREFIX = "detail_";
 
 function isPrimitive(x) {
   return Object(x) !== x;
 }
 
 function sanitize(name) {
-  let sanitizedName = name.replace(/[^a-zA-Z0-9\-_]+/gi, '_');
+  let sanitizedName = name.replace(/[^a-zA-Z0-9\-_]+/gi, '');
   if (name.length > MAX_FIELDNAME_LENGTH) {
     console.warn('Field name ' + name + ' longer than ' + MAX_FIELDNAME_LENGTH + ' will be truncated.');
     sanitizedName = sanitizedName.substring(0, MAX_FIELDNAME_LENGTH);
   }
   return sanitizedName;
 }
 
-function flatten(obj, newObj, prefix) {
+function transformDetails(obj) {
```

Comment:
the method should be named a bit differently I think. Right now, whatever object it gets, it will always prefix it with "detail_". Looking at its name I would suspect that it will transform only some part of the object is gets, namely the details. This looks especially puzzling in the tests, where you call transformDetails(obj) on objects that do not have details. My proposal would be transformToSfxProps(prefix, obj) and the concrete call would be transformToSfxProps(DETAIL_PREFIX, obj).